### PR TITLE
advance kafka offset when deserializer yields no object

### PR DESCRIPTION
**Pull request by prokopowicz**
also removes some tab-indenting in the surrounding code.  

When a custom scheme returns an empty list of object-lists to the partition manager, it should be handled the same as null.  Otherwise, the partition reader will be stuck at that offset.

### CODE REVIEW

File: external/storm-kafka/src/jvm/storm/kafka/PartitionManager.java [hunk 1]
```diff
@@ -136,15 +136,15 @@ public EmitState next(SpoutOutputCollector collector) {
...
-            if (tups != null) {
-		if(_spoutConfig.topicAsStreamId) {
-	            for (List<Object> tup : tups) {
-			collector.emit(_spoutConfig.topic, tup, new KafkaMessageId(_partition, toEmit.offset));
-		    }
-		} else {
-		    for (List<Object> tup : tups) {
-			collector.emit(tup, new KafkaMessageId(_partition, toEmit.offset));
-		    }
+            if ((tups != null) && tups.iterator.hasNext()) {
```
Comment (revans2) [inline external/storm-kafka/src/jvm/storm/kafka/PartitionManager.java]: This doesn't even compile.

>[ERROR] /home/travis/build/apache/storm/external/storm-kafka/src/jvm/storm/kafka/PartitionManager.java:[139,39] cannot find symbol
>
>  symbol:   variable iterator
>
>  location: variable tups of type java.lang.Iterable&lt;java.util.List&lt;java.lang.Object>>

I think it needs to be `tups.iterator().hasNext()`

Comment (prokopowicz) [inline external/storm-kafka/src/jvm/storm/kafka/PartitionManager.java]: sorry :(   I am pushing another commit now

**Comment by revans2** [comment]
@prokopowicz I am +1 for merging this in.  Is there a JIRA associated with this change?

**Comment by HeartSaVioR** [comment]
+1, too.
@prokopowicz 
Since this is a bug, I'd like to record this to JIRA issue before merging. Could you file a JIRA, please?

**Comment by Crim** [comment]
+1 running into this issue as well

**Comment by prokopowicz** [comment]
I issued this JIRA regarding the bug:

https://issues.apache.org/jira/browse/STORM-1094

**Comment by HeartSaVioR** [comment]
Great, I'll merge it.

**Pull request merged**